src/util/dfdom.coffee

```diff
@@ -17,7 +17,9 @@ class DfDomElement
   constructor: (@element) ->
     # @element is a CSS Selector
     if typeof @element is "string"
-      @element = Array.prototype.slice.call document.querySelectorAll @element
+      # check for bad ids (starting with number)
+      selector = if /^#\d/.test @element then  "[id=\"#{@element.substring 1}\"]" else @element
+      @element = Array.prototype.slice.call document.querySelectorAll selector
```

Comment:
Why don't we use

```coffee
(document.querySelectorAll selector).slice
```

directly?

Also, if we are calling `slice` to convert an array-like object into an actual array, we should explicitly indicate it in a line comment because it's not obvious (slice returns a copy and, with no parameters, returns the same elements in the original array).